executor/grant_test.go

```diff
@@ -474,6 +474,38 @@ func TestGrantOnNonExistTable(t *testing.T) {
 	require.NoError(t, err)
 	_, err = tk.Exec("grant Select,Update on test.xx to 'genius'")
 	require.NoError(t, err)
+
+	// issue #29268
+	tk.MustExec("CREATE DATABASE d29268")
+	defer tk.MustExec("DROP DATABASE IF EXISTS d29268")
+	tk.MustExec("USE d29268")
+	tk.MustExec("CREATE USER u29268")
+	defer tk.MustExec("DROP USER u29268")
+
+	// without create privilege
+	err = tk.ExecToErr("GRANT SELECT ON t29268 TO u29268")
+	require.Error(t, err)
+	require.True(t, terror.ErrorEqual(err, infoschema.ErrTableNotExists))
+	fmt.Println(err)
```

Comment:
better remove this line

Reply:
Forgot to remove when developing, I will remove it and check others.